app/models/log.rb

```diff
@@ -98,11 +98,15 @@ def collection_period_open?
   def blank_invalid_non_setup_fields!
     setup_ids = form.setup_sections.flat_map(&:subsections).flat_map(&:questions).map(&:id)
 
+    blank_compound_invalid_non_setup_fields!
+
     errors.each do |error|
       next if setup_ids.include?(error.attribute.to_s)
 
       public_send("#{error.attribute}=", nil)
     end
+
+    valid?
```

Comment:
need to revalidate after blanking fields to remove old errors so we can set log  status correctly